src/autonat/index.ts

```diff
@@ -508,7 +508,7 @@ class DefaultAutoNATService implements Startable {
           // they either told us which address worked/didn't work, or we only sent them one address
           const addr = dialResponse.addr == null ? multiaddrs[0] : multiaddr(dialResponse.addr)
 
-          log('Autonat response for %s is %s', addr, dialResponse.status)
+          log('Autonat response for %s is %s', addr.toString(), dialResponse.status)
```

Comment:
As a rule we shouldn't cause work to be done while logging things.

If we need to call methods on an object to log it properly a [custom formatter](https://github.com/libp2p/js-libp2p-logger/blob/master/src/index.ts#L9-L37) should be created so the functions are only called if logging is enabled.

Reply:
I've opened https://github.com/libp2p/js-libp2p-logger/pull/34 to address this